### PR TITLE
Refactor and test the Command Line Interface

### PR DESCRIPTION
**Pull request by theotherjimmy**

### CODE REVIEW

File: mbed_lstools/main.py [hunk 1]
```diff
@@ -82,165 +82,164 @@ def mbed_lstools_os_info():
...
+def json_by_target_id(mbeds, args):
+    print(json.dumps({m['target_id']: m for m
+                      in mbeds.list_mbeds(unique_names=True,
+                                          read_details_txt=True)},
+                     indent=4, sort_keys=True))
+
+def json_platforms(mbeds, args):
+    platforms = set()
+    for d in mbeds.list_mbeds():
+        platforms |= set([d['platform_name']])
```
Comment (bridadan) [inline mbed_lstools/main.py]:
```python
platforms = set([d['platform_name'] for d in mbeds.list_mbeds()])
```

If you wanted to one line it you could do the above I believe

File: mbed_lstools/main.py [hunk 2]
```diff
@@ -82,165 +82,164 @@ def mbed_lstools_os_info():
...
-                      action="store_true",
-                      help='Prints package version and exits')
-
-    (opts, args) = parser.parse_args()
-    return (opts, args)
-
+    parser = argparse.ArgumentParser()
+    parser.set_defaults(command=print_table)
+
+    commands = parser.add_mutually_exclusive_group()
```
Comment (bridadan) [inline mbed_lstools/main.py]: Hooray! 🎉

File: mbed_lstools/main.py [hunk 3]
```diff
@@ -82,165 +82,164 @@ def mbed_lstools_os_info():
...
-        print(get_mbedls_version())
-
-    else:
-        print(mbeds.get_string(border=not opts.simple, header=not opts.simple))
-
-    logger.debug("Return code: %d", mbeds.ERRORLEVEL_FLAG)
+    logger.debug("Return code: %d", ret_code)
 
-    sys.exit(mbeds.ERRORLEVEL_FLAG)
+    sys.exit(ret_code)
```
Comment (bridadan) [inline mbed_lstools/main.py]: Do you mind removing the ERRORLEVEL_FLAG variable here as well? https://github.com/ARMmbed/mbed-ls/blob/966747a8f6e62260a5bfd5ac3f69ba1b95fd1498/mbed_lstools/lstools_base.py#L68

File: mbed_lstools/main.py [hunk 4]
```diff
@@ -82,165 +82,164 @@ def mbed_lstools_os_info():
...
+        else:
+            logger.error("Could not parse mock from token: '%s'", token)
+
+def list_platforms(mbeds, args):
+    print(mbeds.list_manufacture_ids())
+
+def mbeds_as_json(mbeds, args):
+    print(json.dumps(mbeds.list_mbeds(unique_names=True,
+                                      read_details_txt=True),
+                     indent=4, sort_keys=True))
```
[the remark this answers is not in the document]
Comment (bridadan) [reply mbed_lstools/main.py]: nevermiiiiind

File: mbed_lstools/main.py [hunk 5]
```diff
@@ -82,165 +82,164 @@ def mbed_lstools_os_info():
...
+
+def json_by_target_id(mbeds, args):
+    print(json.dumps({m['target_id']: m for m
+                      in mbeds.list_mbeds(unique_names=True,
+                                          read_details_txt=True)},
+                     indent=4, sort_keys=True))
```
[the remark this answers is not in the document]
Comment (bridadan) [reply mbed_lstools/main.py]: ditto here, ignore!

File: mbed_lstools/main.py (hunk 3 above)
Comment (theotherjimmy) [reply mbed_lstools/main.py]: Yeah, can do.

**Pull request merged**